logstash-core/spec/logstash/runner_spec.rb

```diff
@@ -96,6 +99,32 @@ def run(args); end
     end
   end
 
+  context "--log-in-json" do
+    subject { LogStash::Runner.new("") }
+    let(:logfile) { Stud::Temporary.file }
+    let(:args) { [ "--log-in-json", "-l", logfile.path, "-e", "input {} output{}" ] }
+
+    after do
+      logfile.close
+      File.unlink(logfile.path)
+    end
+
+    before do
+      expect(channel).to receive(:subscribe).with(kind_of(LogStash::Logging::JSON)).and_call_original
+      subject.run(args)
+
+      # Log file should have stuff in it.
+      expect(logfile.stat.size).to be > 0
+    end
+
+    it "should log in valid json. One object per line." do
+      logfile.each_line do |line|
+        expect(line).not_to be_empty
+        expect { JSON.parse(line) }.not_to raise_error
+      end
+    end
+  end
+
```

Comment:
I've updated (see spec_helper) the tests to always force debug level + json logging (at least, as far as I can tell) so that we can catch some problems if they come up.